src/Xrefcheck/Scan.hs

```diff
@@ -153,6 +153,25 @@ readDirectoryWith config scanner root =
         then ""
         else dropTrailingPathSeparator root
 
+-- | Get files that are not ignored by Git or config, but weren't added to Git yet.
+getUntrackedFiles :: ExclusionConfig -> FilePath -> IO [FilePath]
+getUntrackedFiles config root =
+  filter (not . isIgnored)
+  . map (location </>)
+  . L.lines
+  <$> readCreateProcess
+        (shell "git ls-files --others --exclude-standard"){cwd = Just root} ""
+  where
+    isIgnored :: FilePath -> Bool
+    isIgnored = matchesGlobPatterns root $ ecIgnore config
+
+    -- Strip leading "." and trailing "/"
+    location :: FilePath
+    location =
+      if root `equalFilePath` "."
+        then ""
+        else dropTrailingPathSeparator root
```

Comment:
After all, `getUntrackedFiled` half-duplicates `readDirectoryWith`, right?

Reply:
Removed code duplication in last commit

Reply:
In this regard, do you plan to retain the last commit or squash it afterward?

I like how it is put separately, but it contains a lot of changes to the first commit and this is not good.

Reply:
I am going to retain it, but now see problem you mentioned, so I will better move some changes between commits during interactive rebase (I'll write here when done)

Reply:
Cool, will wait for your message.

Reply:
@Martoon-00 done

Reply:
Thank you, looks good :+1: